pkg/reconcile/pipeline/handler/collect/impl_test.go

```diff
@@ -1030,6 +1030,71 @@ var _ = Describe("Integration Collect definitions + items", func() {
 
 })
 
+var _ = Describe("Handle invalid sliceOfMap annotations", func() {
+	var (
+		mockCtrl *gomock.Controller
+	)
+
+	BeforeEach(func() {
+		mockCtrl = gomock.NewController(GinkgoT())
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+	})
+
+	type testCase struct {
+		serviceContent map[string]interface{}
+	}
+
+	It("should set an error condition",
+		func() {
+			tc := testCase{
+				serviceContent: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"annotations": map[string]interface{}{
+							"service.binding/java-maven_port": "path={.status.ports},elementType=sliceOfMaps,sourceKey=name,sourceValue=asdf",
+						},
+					},
+					"status": map[string]interface{}{
+						"ports": []interface{}{
+							map[string]interface{}{"name": "foo", "value": "bar"},
+						},
+					},
+				},
+			}
+
+			ctx := mocks.NewMockContext(mockCtrl)
+			service := mocks.NewMockService(mockCtrl)
+			serviceResource := &unstructured.Unstructured{Object: tc.serviceContent}
+
+			ctx.EXPECT().Services().Return([]pipeline.Service{service}, nil).MinTimes(1)
```

Comment:
`MinTimes(1)` is default if omitted, hence I would remove it.

Reply:
We need `MinTimes(1)` since we're asserting that we call `Services()` at least once, not exactly once.  Default behavior [appears to be](https://stackoverflow.com/questions/51206450/testing-with-gomock-returns-error-expected-call-has-already-been-called-the-max) asserting that it gets called exactly once.